pkg/gofr/datasource/kv-store/badger/badger.go

```diff
@@ -0,0 +1,125 @@
+package badger
+
+import (
+	"context"
+	"errors"
+	"github.com/dgraph-io/badger/v4"
+)
+
+type Configs struct {
+	DirPath string
+}
+
+type client struct {
+	db      *badger.DB
+	configs Configs
+	logger  Logger
+	metrics Metrics
+}
+
+func New(configs Configs) *client {
+	return &client{configs: configs}
+}
+
+// UseLogger sets the logger for the MongoDB client which asserts the Logger interface.
+func (c *client) UseLogger(logger any) {
+	if l, ok := logger.(Logger); ok {
+		c.logger = l
+	}
+}
+
+// UseMetrics sets the metrics for the MongoDB client which asserts the Metrics interface.
+func (c *client) UseMetrics(metrics any) {
+	if m, ok := metrics.(Metrics); ok {
+		c.metrics = m
+	}
+}
+
+// Connect establishes a connection to MongoDB and registers metrics using the provided configuration when the client was Created.
+func (c *client) Connect() {
+	db, err := badger.Open(badger.DefaultOptions(c.configs.DirPath))
+	if err != nil {
+		c.logger.Errorf("failed to initialise badgerDB : %v", err)
+	}
+
+	c.db = db
+}
+
+func (c *client) Get(_ context.Context, key string) (string, error) {
+	var value []byte
+
+	txn := c.db.NewTransaction(false)
```

Comment:
Does a getter really require a transaction?

I may understand to use the transaction (storing it in the context could help here), but not a new one.

And not commit one

Reply:
Its required as per the badgerDB documentation, to keep the data consistent.

Reply:
Maybe it requires a comment to explain it